### PR TITLE
chore(editor-content): implement dynamic columns in relationship fields

### LINKED ISSUE #30923: Relationship Field: Listing Content with Dynamic Columns
### Parent Issue:

https://github.com/dotCMS/core/issues/30495

## User Story:

As a user, I want to view related content in a dialog when browsing through items, so that I can explore additional resources without leaving my current page.

Screenshot 2024-12-11 at 3 39 33 PM

### 1. Dynamic Columns

- The mandatory columns are: Type, Title, Step, and Language.
- The columns displayed in the dialog should be fetched dynamically based on the content type.
- Use the `/api/v3/contenttype/<id>/fields/allfields?filter=SHOW_IN_LIST` endpoint to retrieve the fields to be displayed.
- The response should determine which fields (e.g., Title, Type, URL, Publish Date) appear as columns in the dialog.

### 2. Content Display

- Utilize the `/api/content/_search` endpoint to retrieve content dynamically.
- The dialog should display a list of related content items, dynamically binding the retrieved fields to the corresponding columns.
- The display should include thumbnails for items where applicable.
- When there are many columns, implement horizontal scrolling, ensuring the checkbox and button columns are sticky for better usability.

### 3. Error Handling

- If there is no related content, the dialog should display a user-friendly message (e.g., "No related content available.").
- If content fails to load due to a technical error, a retry option should be provided.

### PR DESCRIPTION
**Pull request by nicobytes**
### Parent Issue

#30923 

### Proposed Changes

This pull request includes several changes across different files to introduce a new service, update styling, and improve existing components. The most important changes include adding a new `DotFieldService`, updating SCSS variables and styles, and enhancing the `dot-select-existing-content` component.

### Introduction of `DotFieldService`:

* Added `DotFieldService` to handle fetching fields for a specific content type with optional filtering. (`core-web/libs/data-access/src/lib/dot-field/dot-field.service.ts`)
* Created unit tests for `DotFieldService` to verify its functionality. (`core-web/libs/data-access/src/lib/dot-field/dot-field.service.spec.ts`)
* Exported `DotFieldService` in the index file. (`core-web/libs/data-access/src/index.ts`)

### SCSS Variables and Styles:

* Updated SCSS variables to use CSS custom properties for gray shades. (`core-web/libs/dotcms-scss/shared/_colors.scss`)
* Added new CSS custom properties for border radius and primeflex colors. (`core-web/libs/dotcms-scss/angular/styles.scss`, `core-web/libs/dotcms-scss/shared/_colors.scss`) [[1]](diffhunk://#diff-19fbe1ab2e0dcad6949b44c72556e4893c233c54263211aa055d0034241fcb5fR43-R44) [[2]](diffhunk://#diff-a673541496e37f0c8d73eef81a2026235e3d621daf64af8bf35a97881cb2dce5R430-R451)
* Changed table highlight background color. (`core-web/libs/dotcms-scss/angular/dotcms-theme/components/_table.scss`)

### Enhancements to `dot-select-existing-content` Component:

* Enhanced the component to display error messages and handle empty data scenarios. (`core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/dot-select-existing-content.component.html`)
* Updated the component to use dynamic columns and improved table layout. (`core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/dot-select-existing-content.component.html`, `core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/dot-select-existing-content.component.scss`) [[1]](diffhunk://#diff-ba794904cc95635d9efca07b7e2bfeb948b5b0bb0c8e1e97c777c615398ba011R2) [[2]](diffhunk://#diff-778cca685214b49b5445efc5613609b68214a24052f4ed77408caeb0c9e6ffb2L5-L13)
* Added a required input for content ID to the component. (`core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/dot-select-existing-content.component.ts`)
* Updated unit tests to mock the `RelationshipFieldService` and reflect changes in the component. (`core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/dot-select-existing-content.component.spec.ts`)

### Checklist
- [x] Tests
- [x] Translations
- [x] Security Implications Contemplated (add notes if applicable)

This PR fixes: #30923

### CODE REVIEW

**Comment by nicobytes** [comment]
QA Notes:

1. A relationship without any associated content or data.

https://github.com/user-attachments/assets/62661c3b-551b-4d08-924f-b63e77e1a7fd

2. Show related content fields based on “Show in List”

https://github.com/user-attachments/assets/b4c48d9c-e83e-497b-adf5-0a542b34b23d

3. Display the title as the ID when no text field is available.

https://github.com/user-attachments/assets/0b2b2a8f-3037-47f0-a009-26d1c0f44b21

4. Manage multiple fields using horizontal scrolling and sticky columns

https://github.com/user-attachments/assets/9bcd6556-e903-47cc-9e3c-7fe0f149785f

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/dot-select-existing-content.component.html [hunk 1]
```diff
@@ -21,93 +22,98 @@
...
+                                <th
+                                    class="max-w-12rem"
+                                    scope="col"
+                                    [pSortableColumn]="column.field">
+                                    <span class="capitalize">{{ column.header }}</span>
+                                    <p-sortIcon [field]="column.field" />
+                                </th>
+                            }
+                            <th scope="col" pFrozenColumn alignFrozen="right" [frozen]="true">
+                                Menu
```
Comment (hmoreras) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/dot-select-existing-content.component.html:96]: translate this ?

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/store/existing-content.store.ts [hunk 2]
```diff
@@ -56,15 +54,34 @@ export const ExistingContentStore = signalStore(
...
-                                error: () => patchState(store, { status: ComponentStatus.ERROR })
+                                next: ([columns, data]) => {
+                                    patchState(store, {
+                                        columns,
+                                        data,
+                                        status: ComponentStatus.LOADED
+                                    });
+                                },
+                                error: () =>
+                                    patchState(store, {
```
Comment (hmoreras) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/store/existing-content.store.ts:80]: should we use here the `dotHttpErrorManagerService` to show a dialog to the user with the actual error ?

**Review by hmoreras — APPROVED** [review]

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.html [hunk 3]
```diff
@@ -88,5 +89,6 @@
 </p-table>
 
 <dot-select-existing-content
+    [contentId]="$contentId()"
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.html:92]: what is contentID? inode or identifier?

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/dot-select-existing-content.component.ts [hunk 4]
```diff
@@ -75,6 +83,12 @@ export class DotSelectExistingContentComponent {
      */
     $isApplyDisabled = computed(() => this.$selectedItems().length === 0);
 
+    /**
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/dot-select-existing-content.component.ts:86]: this need to be inode or identifier.

Comment (oidacra) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/dot-select-existing-content.component.ts:86]: chage it to use contentTypeId direct.

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.html (hunk 3 above)
Comment (oidacra) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.html:92]: the correct name needs to be contentTypeId.

If we leave it as contentId, when we touch this again we will start looking to see if that is an inode or an identifier, in the end it is really a contentTypeId

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/services/relationship-field.service.ts [hunk 5]
```diff
@@ -1,85 +1,159 @@
...
+                title: item.title || item.identifier,
+                language: languages[item.languageId] || '',
+                modDate: formatDate(item.modDate, 'short', 'en-US')
+            };
+
+            return relationshipItem;
+        });
     }
 
     #getRandomState(): { label: string; styleClass: string } {
```
Comment (oidacra) [inline core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/services/relationship-field.service.ts:159]: No need delete this?

Comment (nicobytes) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/services/relationship-field.service.ts]: Removed

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/dot-select-existing-content.component.ts (hunk 4 above)
Comment (nicobytes) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/dot-select-existing-content.component.ts:86]: I agree, fixed!

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.html (hunk 3 above)
Comment (nicobytes) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/dot-edit-content-relationship-field.component.html]: Fixed!

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/store/existing-content.store.ts (hunk 2 above)
Comment (nicobytes) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/store/existing-content.store.ts:80]: I applied this `dotHttpErrorManagerService` in the `RelationshipFieldService`

File: core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/dot-select-existing-content.component.html (hunk 1 above)
Comment (nicobytes) [reply core-web/libs/edit-content/src/lib/fields/dot-edit-content-relationship-field/components/dot-select-existing-content/dot-select-existing-content.component.html]: Fixed

**Review by oidacra — APPROVED** [review]

**Pull request merged**